sorter.go

```diff
@@ -63,6 +65,9 @@ func (l keyList) Less(i, j int) bool {
 		if ai != bi {
 			return ai < bi
 		}
+		if ar[i] == br[i] {
+			continue
```

Comment:
I suspect this may not be quite right. The original code always returned after comparing numbers, but this now means that we can carry on instead. At the least, this introduces quadratic behaviour which might be a problem for long keys with many decimal digits in. I'll need to think some more about this.